server/game/cards/06-CotE/PrudentChallenger.js

```diff
@@ -0,0 +1,36 @@
+const DrawCard = require('../../drawcard.js');
+
+class PrudentChallenger extends DrawCard {
+    setupCardAbilities() {
+        this.action({
+            title: 'Initiate a military duel',
+            initiateDuel: context => ({
+                type: 'military',
+                resolutionHandler: (winner, loser) => this.resolutionHandler(context, winner, loser)
+            })
+        });
+    }
+
+    resolutionHandler(context, winner, loser) {
+        if(loser) {
+            this.game.addMessage('{0} wins the duel and {1} loses', winner, loser);
+            if(loser.attachments.size() > 0) {
+                this.game.promptWithHandlerMenu(context.player, {
+                    activePromptTitle: 'Choose an attachment to discard',
+                    context: context,
+                    cards: loser.attachments.sortBy(card => card.name),
+                    cardHandler: card => {
+                        this.game.addMessage('{0} chooses to discard {1}', context.player, card);
+                        this.game.applyGameAction(context, { discardCard: card });
+                    }
+                });
+            }
```

Comment:
a) I think we should be using a SelectCardPrompt, not a HandlerMenuPrompt here
b) This is a perfect example of the kind of card which is going to cause major headaches with Hantei - I suspect that we need to restructure duels to have a gameAction rather than a resolutionHandler....

Reply:
a) I have changed to a select prompt as requested
b) Yes a winnerGameAction and a loserGameAction would be ideal and while we are at it I think we could do with standardising the chat messages for duels, maybe something like `Duel Result - Winner: Prudent Challenger, Loser: Doji Whisperer`